core/sail/spin/src/test/java/org/eclipse/rdf4j/spin/functions/SpinFunctionTest.java

```diff
@@ -0,0 +1,275 @@
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.rdf4j.query.BooleanQuery;
+import org.eclipse.rdf4j.query.QueryLanguage;
+import org.eclipse.rdf4j.query.QueryResults;
+import org.eclipse.rdf4j.query.TupleQuery;
+import org.eclipse.rdf4j.query.TupleQueryResult;
+import org.eclipse.rdf4j.query.algebra.evaluation.function.TupleFunction;
+import org.eclipse.rdf4j.query.resultio.QueryResultIO;
+import org.eclipse.rdf4j.query.resultio.TupleQueryResultFormat;
+import org.eclipse.rdf4j.repository.RepositoryException;
+import org.eclipse.rdf4j.repository.sail.SailQuery;
+import org.eclipse.rdf4j.repository.sail.SailRepository;
+import org.eclipse.rdf4j.repository.sail.SailRepositoryConnection;
+import org.eclipse.rdf4j.rio.RDFFormat;
+import org.eclipse.rdf4j.sail.NotifyingSail;
+import org.eclipse.rdf4j.sail.inferencer.fc.DedupingInferencer;
+import org.eclipse.rdf4j.sail.inferencer.fc.ForwardChainingRDFSInferencer;
+import org.eclipse.rdf4j.sail.memory.MemoryStore;
+import org.eclipse.rdf4j.sail.spin.SpinSail;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Sets;
+
+/**
+ * Unit tester for the property functions.</br>
+ * <h2>How does it work?</h2></br>
+ * <ul>
+ * <li>The functions should be embedded into a SPARQL query.</li>
+ * <li>The query might be any type: graph, tuple or boolean.</li>
+ * <li>The file with sparql query should be written in <code>src/test/resources/functions/</code>
+ * directory.</li>
+ * </ul>
+ * 
+ * @author Jacek Grzebyta
+ */
+@RunWith(Parameterized.class)
+public class SpinFunctionTest {
+
+	private static Logger log = LoggerFactory.getLogger(SpinFunctionTest.class);
+
+	private SailRepository repository;
+
+	private SailRepositoryConnection connection;
+
+	public class TestingCase {
+
+		public Path testFile;
+
+		public Path resultFile;
+	}
+
+	public static final String TESTS_ROOT = SpinFunctionTest.class.getResource("/functions/").getFile();
+
+	// Extensions for different file formats
+	public static final String SPARQL_EXT = ".rq";
+
+	public static final String SPARQL_RES_EXT = ".srx";
+
+	@Parameters
+	public static Collection<Object[]> loadSparqls()
+		throws Exception
+	{
+		List<Object[]> sparqls = new ArrayList<>();
+
+		List<Path> functions = scanTests(Paths.get(TESTS_ROOT));
```

Comment:
This should be done using a manifest, (see the SPARQL/RDF compliance testsuites for examples) rather than a file scan.